lasagne/layers/conv.py

```diff
@@ -248,8 +253,8 @@ def get_output_for(self, input, input_shape=None, **kwargs):
                                       image_shape=input_shape,
                                       filter_shape=self.get_W_shape(),
                                       border_mode='full')
-            shift = (self.filter_size[0] - 1) // 2
-            conved = conved[:, :, shift:input.shape[2] + shift]
+            crop = self.filter_size[0] // 2
+            conved = conved[:, :, crop:-crop]
```

Comment:
Should be `conved[:, :, crop:-crop or None]`, otherwise it will fail for `self.filter_size[0] == 1`!